src/UI/Implementation/Component/Listing/DescriptiveList/DescriptiveList.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Listing\DescriptiveList;
+
+use ILIAS\UI\Component as C;
+
+/**
+ * Class DescriptiveList
+ * @package ILIAS\UI\Implementation\Component\Listing\DescriptiveList
+ */
+class DescriptiveList implements C\Listing\DescriptiveList {
+
+	/**
+	 * @var	array
+	 */
+	private  $items;
+
+	/**
+	 * @inheritdoc
+	 */
+	public function __construct(array $items) {
```

Comment:
Type hinting will not work, as PHP lacks TypeParams and i can't hint on, say, `array<string,string|component>`. So we could only hint on array. On top of that, we could use something like ComponentHelper::checkArgListElements. It seems as we would need a new method for your use case, that checks keys and values as well. I could write one on monday.